src/clojure_tictactoe/cli/input/input_getter.clj

```diff
@@ -13,25 +14,38 @@
   (get-user-input)
   nil)
 
-(defn parse-move-input [input]
-  (let [formatted-input (read-string input)]
-    (let [valid-numeral (and
-                          (number? formatted-input)
-                          (< formatted-input 9))]
-      (when valid-numeral
-        formatted-input))))
+(defn parse-move-input [input choice-limit]
+  (let [formatted-input (read-string input)
+        valid-numeral (and
+                        (number? formatted-input)
+                        (< formatted-input choice-limit))]
+    (when valid-numeral
+      formatted-input)))
 
-(defn get-player-choice []
+(defn get-player-marker [player]
+  (instructions-printer/print-marker-instructions player)
+  (get-user-input))
+
+(defn get-player-choice [board-length]
   (println (str "Which space would you like to mark?"))
-  (if-let [choice (parse-move-input (get-user-input))]
+  (if-let [choice (parse-move-input (get-user-input) board-length)]
     choice
     (do (println "Sorry, looks like that's not possible, try again?")
-        (recur))))
+        (recur board-length))))
 
 (defn get-player-move [board]
+  (let [move (get-player-choice (count board))]
+    (if (board/space-is-open? move board)
+      move
+      (do (println "Sorry, that looks taken, try again")
+          (recur board)))))
+
+(defn get-game-type [options]
+  (println instructions-printer/game-choice-message)
+  (instructions-printer/print-stringified-options options)
   (loop []
-    (let [move (get-player-choice)]
-      (if (board/space-is-open? move board)
-        move
-        (do (println "Sorry, that looks taken, try again")
-            (recur))))))
+    (println instructions-printer/game-choice-request)
+    (let [choice-limit (+ (count options) 1)]
+      (if-let [choice (parse-move-input (get-user-input) choice-limit)]
```

Comment:
I thought I left a comment about this on your last PR, but I think you really need to rename `parse-move-input` because here, it is definitely not parsing move input.

Reply:
You did. And I agree.